src/d/actor/d_a_shutter2.cpp

```diff
@@ -5,75 +5,230 @@
+    }
+    else {
+        m30C = 2;
+    }
+    mbIsSwitch = fopAcM_isSwitch(this, mSwitchNo);
+    set_mtx();
+    if (m_open_ev_name[mType] != NULL) {
+        mOpenEventIdx = dComIfGp_evmng_getEventIdx(m_open_ev_name[mType], 0xff);
+    }
+    if (m_close_ev_name[mType] != NULL) {
+        mCloseEventIdx = dComIfGp_evmng_getEventIdx(m_close_ev_name[mType], 0xff);
+    }
+    return TRUE;
 }
 
 /* 00000350-0000041C       .text _create__12daShutter2_cFv */
 s32 daShutter2_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daShutter2_c);
+    mType = 0;
+    int result = dComIfG_resLoad(&mPhs, m_arcname[mType]);
+    if (result == cPhs_COMPLEATE_e) {
+        if (MoveBGCreate(m_arcname[mType], m_dzbidx[mType], NULL, m_heapsize[mType]) == 5) {
+            result = cPhs_ERROR_e;
+        }
+        else {
+            return;
+        }
+    }
+    return result;
 }
 
 /* 0000041C-000004AC       .text set_mtx__12daShutter2_cFv */
 void daShutter2_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    MTXCopy(mDoMtx_stack_c::get(), mMtx);
+    return;
 }
 
 /* 000004AC-00000528       .text Execute__12daShutter2_cFPPA3_A4_f */
-void daShutter2_c::Execute(float(**)[3][4]) {
-    /* Nonmatching */
+BOOL daShutter2_c::Execute(Mtx** i_this) {
+    demo();
+    set_mtx();
+    mbIsSwitch = fopAcM_isSwitch(this, mSwitchNo);
+    mbIsNearEnemy = fopAcM_myRoomSearchEnemy(fopAcM_GetRoomNo(this)) == NULL;
+    *i_this = &mMtx;
+    return TRUE;
 }
 
 /* 00000528-00000698       .text shutter_move__12daShutter2_cFv */
 void daShutter2_c::shutter_move() {
-    /* Nonmatching */
+    float fVar3;
+
+    static char* action_table[4] = {"WAIT", "OPEN", "CLOSE", "OPEN_INIT"};
+    int actionIndex = dComIfGp_evmng_getMyActIdx(mStaffId, action_table, ARRAY_SIZE(action_table), FALSE, 0);
+
+    float maxVel = m_max_speed[mType];
+    float minVel = m_min_speed[mType];
+    
+    switch (actionIndex) {
+        case 0: //WAIT
+        {
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;   
+        }
+        case 3: //OPEN_INIT
+        {
+            fopAcM_seStart(this, JA_SE_OBJ_WDUN_R04_STR_OP, 0);
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+        case 1: //OPEN
+        {
+            fVar3 = cLib_addCalc(&current.pos.y, home.pos.y + 350.0f, 0.1f, maxVel, minVel);
+            if (fVar3 == 0.0f){
+                dComIfGp_evmng_cutEnd(mStaffId);
+            }
+            return;
+
+        }
+        case 2: //CLOSE
+        {
+            fVar3 = cLib_addCalc(&current.pos.y, home.pos.y, 0.1f, maxVel, minVel);
+            if (fVar3 == 0.0f){
+                dComIfGp_evmng_cutEnd(mStaffId);
+            }
+            return;
+        }
+    }
+    dComIfGp_evmng_cutEnd(mStaffId);
```

Comment:
I think all the `return` statements inside this switch are probably supposed to be `break`s. They happen to look identical in this function since the switch is at the end of the function, but I've seen other places where cutEnd went inside a default case along these lines:
```suggestion
            break;
        }
        default:
        {
            dComIfGp_evmng_cutEnd(mStaffId);
            break;
        }
    }
```